### PR TITLE
Move TCL test unit/expire to Go case 

### PR DESCRIPTION
**Pull request by maochongxin**
#856

### CODE REVIEW

File: tests/gocase/unit/expire/expire_test.go [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+func TestExpire(t *testing.T) {
+	svr := util.StartServer(t, map[string]string{})
+	defer svr.Close()
+
+	ctx := context.Background()
+	rdb := svr.NewClient()
+	defer func() { require.NoError(t, rdb.Close()) }()
+
+	t.Run("EXPIRE - set timeouts multiple times", func(t *testing.T) {
+		rdb.Set(ctx, "x", "foobar", 0)
```
Comment (tanruixiang) [inline tests/gocase/unit/expire/expire_test.go]:
```suggestion
                require.NoError(t, rdb.Set(ctx, "x", "foobar", 0).Err())
```

File: tests/gocase/unit/expire/expire_test.go [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+		require.Equal(t, 1, rdb.Expire(ctx, "x", 5*time.Second).Val())
+		ttl := rdb.TTL(ctx, "x").Val()
+		require.LessOrEqual(t, 5, ttl)
+		require.GreaterOrEqual(t, 4, ttl)
+		require.Equal(t, 1, rdb.Expire(ctx, "x", 10*time.Second).Val())
+		require.Equal(t, 10*time.Second, rdb.TTL(ctx, "x").Val())
+		rdb.Expire(ctx, "x", 2*time.Second)
```
Comment (tanruixiang) [inline tests/gocase/unit/expire/expire_test.go]:
```suggestion
                require.NoError(t, rdb.Expire(ctx, "x", 2*time.Second).Err())
```

File: tests/gocase/unit/expire/expire_test.go [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+
+	t.Run("EXPIRE - It should be still possible to read 'x'", func(t *testing.T) {
+		require.Equal(t, "foobar", rdb.Get(ctx, "x").Val())
+	})
+
+	t.Run("EXPIRE - write on expire should work", func(t *testing.T) {
+		rdb.Del(ctx, "x")
+		rdb.LPush(ctx, "x", "foo")
+		rdb.Expire(ctx, "x", 1000*time.Second)
+		rdb.LPush(ctx, "x", "bar")
```
Comment (tanruixiang) [inline tests/gocase/unit/expire/expire_test.go]:
```suggestion
                require.NoError(t, rdb.Del(ctx, "x").Err())
                require.NoError(t, rdb.LPush(ctx, "x", "foo").Err())
                require.NoError(t, rdb.Expire(ctx, "x", 1000*time.Second).Err())
                require.NoError(t, rdb.LPush(ctx, "x", "bar").Err())
```

File: tests/gocase/unit/expire/expire_test.go [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+		require.Equal(t, []string{"bar", "foo"}, rdb.LRange(ctx, "x", 0, -1).Val())
+	})
+
+	t.Run("EXPIREAT - Check for EXPIRE alike behavior", func(t *testing.T) {
+		rdb.Del(ctx, "x")
+		rdb.Set(ctx, "x", "foo", 0)
+		rdb.ExpireAt(ctx, "x", time.Now().Add(15*time.Second))
```
Comment (tanruixiang) [inline tests/gocase/unit/expire/expire_test.go]:
```suggestion
                require.NoError(t, rdb.Del(ctx, "x").Err())
                require.NoError(t, rdb.Set(ctx, "x", "foo", 0).Err())
                require.NoError(t, rdb.ExpireAt(ctx, "x", time.Now().Add(15*time.Second)).Err())
```

**Review by tanruixiang — CHANGES_REQUESTED** [review]
Thanks for your contribution. When using `rdb`, we need to judge the returned error.

File: tests/tcl/tests/test_helper.tcl [hunk 5]
```diff
@@ -37,7 +37,7 @@ set ::all_tests {
     unit/type/list
     unit/type/zset
     unit/type/stream
-    unit/expire
+    unit/multi
```
Comment (PragmaTwice) [inline tests/tcl/tests/test_helper.tcl:40]: be careful while solving conflicts :)

Comment (PragmaTwice) [inline tests/tcl/tests/test_helper.tcl]:
```suggestion
```

Comment (PragmaTwice) [reply tests/tcl/tests/test_helper.tcl]: be careful while solving conflicts :)

File: tests/tcl/tests/unit/expire.tcl [hunk 6]
```diff
@@ -1,206 +0,0 @@
...
-        r set s c
-        r set foo b
-        lsort [r keys *]
-    } {a e foo s t}
-
-    test {EXPIRE with empty string as TTL should report an error} {
-        r set foo bar
-        catch {r expire foo ""} e
-        set e
-    } {*not an integer*}
```
Comment (PragmaTwice) [inline tests/tcl/tests/unit/expire.tcl:205]: is there corresponding go code for this case?

Comment (maochongxin) [reply tests/tcl/tests/unit/expire.tcl:205]: At first I thought that the `go-redis Expire` method would have type checking, so it was not implemented, it has been added

File: tests/gocase/unit/expire/expire_test.go [hunk 7]
```diff
@@ -0,0 +1,224 @@
...
+		//require.NoError(t, rdb.Set(ctx, "key1", "a", 500*time.Millisecond).Err())
+		//require.NoError(t, rdb.Set(ctx, "key2", "a", 500*time.Millisecond).Err())
+		//require.NoError(t, rdb.Set(ctx, "key3", "a", 500*time.Millisecond).Err())
+		require.NoError(t, rdb.Do(ctx, "DBSIZE", "scan").Err())
+		time.Sleep(100 * time.Millisecond)
+		require.Equal(t, int64(3), rdb.DBSize(ctx).Val())
+		time.Sleep(2000 * time.Millisecond)
+		require.NoError(t, rdb.Do(ctx, "DBSIZE", "scan").Err())
+		time.Sleep(100 * time.Millisecond)
+		require.Equal(t, int64(0), rdb.DBSize(ctx).Val())
```
Comment (git-hulk) [inline tests/gocase/unit/expire/expire_test.go]:
```suggestion
		require.NoError(t, rdb.Do(ctx, "DBSIZE", "scan").Err())
		time.Sleep(100 * time.Millisecond)
		require.EqualValues(t, 3, rdb.DBSize(ctx).Val())
		time.Sleep(2000 * time.Millisecond)
		require.NoError(t, rdb.Do(ctx, "DBSIZE", "scan").Err())
		time.Sleep(100 * time.Millisecond)
		require.EqualValues(t, 0, rdb.DBSize(ctx).Val())
```

File: tests/gocase/unit/expire/expire_test.go [hunk 8]
```diff
@@ -0,0 +1,224 @@
...
+	})
+
+	t.Run("Redis should actively expire keys incrementally", func(t *testing.T) {
+		require.NoError(t, rdb.FlushDB(ctx).Err())
+		require.NoError(t, rdb.Do(ctx, "PSETEX", "key1", 500, "a").Err())
+		require.NoError(t, rdb.Do(ctx, "PSETEX", "key2", 500, "a").Err())
+		require.NoError(t, rdb.Do(ctx, "PSETEX", "key3", 500, "a").Err())
+		//require.NoError(t, rdb.Set(ctx, "key1", "a", 500*time.Millisecond).Err())
+		//require.NoError(t, rdb.Set(ctx, "key2", "a", 500*time.Millisecond).Err())
+		//require.NoError(t, rdb.Set(ctx, "key3", "a", 500*time.Millisecond).Err())
```
Comment (PragmaTwice) [inline tests/gocase/unit/expire/expire_test.go]:
```suggestion
```

**Review by git-hulk — COMMENTED** [review]
@maochongxin Can you also add the below missing test cases:

* https://github.com/apache/incubator-kvrocks/blob/1fd312c183371cd93ddc32087d0113aaaf5ee9d3/tests/tcl/tests/unit/expire.tcl#L39
* https://github.com/apache/incubator-kvrocks/blob/1fd312c183371cd93ddc32087d0113aaaf5ee9d3/tests/tcl/tests/unit/expire.tcl#L75

File: tests/gocase/unit/expire/expire_test.go (hunk 7 above)
Comment (maochongxin) [reply tests/gocase/unit/expire/expire_test.go]: done

> @maochongxin Can you also add the below missing test cases:
> 
> * https://github.com/apache/incubator-kvrocks/blob/1fd312c183371cd93ddc32087d0113aaaf5ee9d3/tests/tcl/tests/unit/expire.tcl#L39
> * https://github.com/apache/incubator-kvrocks/blob/1fd312c183371cd93ddc32087d0113aaaf5ee9d3/tests/tcl/tests/unit/expire.tcl#L75

**Review by git-hulk — DISMISSED** [review]

**Review by tisonkun — COMMENTED** [review]
Thank you! LGTM. I'm pushing several changes to use tidy assertions.

**Review by tisonkun — APPROVED** [review]

**Review by tanruixiang — APPROVED** [review]

**Review by git-hulk — APPROVED** [review]

**Comment by maochongxin** [comment]
orz, thanks masters, I will continue to contribute

**Review by PragmaTwice — APPROVED** [review]

**Comment by tisonkun** [comment]
@maochongxin thank you!

**Pull request merged**